lstchain/reco/dl0_to_dl1.py

```diff
@@ -92,19 +94,33 @@ def get_dl1(calibrated_event, telescope_id, dl1_container=None, custom_config={}
 
     signal_pixels = cleaning_method(camera, image, **cleaning_parameters)
 
+
     if image[signal_pixels].sum() > 0:
-        hillas = hillas_parameters(camera[signal_pixels], image[signal_pixels])
+
+        # check the number of islands 
+        num_islands, island_labels = number_of_islands(camera, signal_pixels)
+        intensity_of_island = np.zeros(num_islands+1)
+
+        for iisland in range(1, num_islands+1):
+            intensity_of_island[iisland] = np.sum(image[island_labels==iisland])
+
+        max_island_label = np.argmax(intensity_of_island)
+        signal_pixels_main = deepcopy(signal_pixels)
+        signal_pixels_main[island_labels!=max_island_label] = False
+
+        hillas = hillas_parameters(camera[signal_pixels_main], image[signal_pixels_main])
+
         # Fill container
         dl1_container.fill_mc(calibrated_event)
         dl1_container.fill_hillas(hillas)
         dl1_container.fill_event_info(calibrated_event)
         dl1_container.set_mc_core_distance(calibrated_event, telescope_id)
         dl1_container.set_mc_type(calibrated_event)
-        dl1_container.set_timing_features(camera[signal_pixels],
-                                          image[signal_pixels],
-                                          pulse_time[signal_pixels],
+        dl1_container.set_timing_features(camera[signal_pixels_main],
+                                          image[signal_pixels_main],
+                                          pulse_time[signal_pixels_main],
                                           hillas)
-        dl1_container.set_leakage(camera, image, signal_pixels)
+        dl1_container.set_leakage(camera, image, signal_pixels_main)
         dl1_container.set_n_islands(camera, signal_pixels)
```

Comment:
in response to my previous comment, to avoid recomputing the number of islands, you could set it manually :
`dl1_container.n_islands = num_islands`
But the operation cost is a low and you could argue that it is more clear so as you prefer.